xds/internal/client/xds.go

```diff
@@ -769,3 +780,130 @@ func parseEDSRespProto(m *v3endpointpb.ClusterLoadAssignment) (EndpointsUpdate,
 	}
 	return ret, nil
 }
+
+// processAllResources unmarshal and validate the resources, populate the
+// provided ret (a map), and return metadata and error.
```

Comment:
Nit: the verbs need to end in `s`.
unmarshals, validates, populates, returns.

Reply:
Done